site/cypress/support/commands.js

```diff
@@ -0,0 +1,75 @@
+// ***********************************************
+// commands.js creates various custom commands and overwrite
+// existing commands.
+//
+// For more comprehensive examples of custom
+// commands please read more here:
+// https://on.cypress.io/custom-commands
+// ***********************************************
+//
+//
+// -- This is a parent command --
+// Cypress.Commands.add("login", (email, password) => { ... })
+//
+//
+// -- This is a child command --
+// Cypress.Commands.add("drag", { prevSubject: 'element'}, (subject, options) => { ... })
+//
+//
+// -- This is a dual command --
+// Cypress.Commands.add("dismiss", { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })
+
+
+//Adding a cypress function requires each function to yield a value instead of returning it directly
+//These functions are defined in normal JS and can be imported instead
+
+
+/**
+* Generate a 3 letter semester code e.g s21, f20 based on today's data
+* This functions the same as the submitty python util's get_current_semester
+*
+* @returns {String}
+*/
+export function getCurrentSemester(){
+	const today = new Date();
+	const year = today.getFullYear().toString().slice(2,4);	//get last two digits
+	const semester = ((today.getMonth() + 1) < 7) ? "s" : "f";	//first half of year 'spring' rest is fall
+
+	return semester + year;
+}
+
+/**
+* Build a courseURL based on an array of 'parts', e.g [foo, bar] -> courses/s21/foo/bar
+* 
+* @param {String[]} [parts=[]] array of parts to string together
+* @param {Boolean} [include_base=false] weather to include the url base (http://localhost:1501/) or not
+* @returns {String}
+*/
+export function buildUrl(parts = [], include_base = false){
```

Comment:
I think it might be good to build this function into a custom `Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })`? This would simplify writing the tests if people could just write something like `cy.visit(['sample', 'gradeable', 'open_homework'])` and it automatically gets morphed to `originalFn('/s20/sample/gradeable/open_homework')`

Reply:
Sounds good
I think a good design would be to check if the user passed in an array, if so expand the parts to a normal URL, if a string was passed in assume its a URL and the user has a specific path they want to visit, for example, if they wanted a to test a specific parameter list